docs/src/single-validator-manager.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Single Validator Stake Pool Manager Program
+---
+
+A work-in-progress program for permissionless management of stake pools that are
+only delegated to one validator.
+
+## Overview
+
+The single-validator manager program defines a canonical stake pool for any
+validator on the network, which can be created by anyone.
+
+The program manages the "delegation strategy" of the pool, which is to maximize
+the amount of stake on one validator, while maintaining a small amount of SOL
+in the reserve for small stakers.
+
+## Fees
+
+Pools are configured with a SOL deposit fee of 8 basis points, which corresponds
+to the rewards received over two epochs. This way, no user can deposit SOL and
+immediately withdraw a stake account at the end of an epoch to steal rewards
+from the pool.
+
+All other fees are 0 so that pools mimic the reward structure of native staking.
+
+The program also exposes a permissionless `BurnFees` instruction to allow anyone
+to burn the pool tokens in the fee account, which increases the scarcity of all
+other pool tokens.
+
+## Operation
+
+### Create Pool
+
+The `CreateStakePool` instruction allows anyone to create the canonical stake
+pool for a validator, at the cost of ~0.0124 SOL for account rent-exemption of
+the stake pool, validator list, reserve, mint, and fee account.
+
+After the pool is created, the reserve requires at least stake rent-exemption plus
+the minimum delegation amount to actually add the validator to the pool, which
+costs 1.0028288 SOL, assuming a minimum delegation of 1 SOL. The user receives
+pool tokens for this deposit.
+
+After the SOL deposit, the `AddValidatorToPool` instruction on the pool management
```

Comment:
similar to what we discovered with `AddValidatorToPool` and `CreateValidatorStake` on the stake pool program, is there ever a case where we would need to call `AddValidatorToPool` separately from `CreateStakePool`? If not, should we just merge these 2 instructions?

Reply:
Any existing stake pool can be reassigned to be managed by `PDA(validator, program_id)`, in which case they must be able to add that validator and remove any others.

Also, starting with #3714, the reserve is used for adding validators, so you'd also need to do a `DepositSol` on the stake pool in between in order to add the validator. It seems nicer to keep them all separated and have a good client library